packages/devtools/test/utils_test.dart

```diff
@@ -20,5 +20,41 @@ void main() {
       expect(printMb(1000 * MB, 1), '1000.0');
       expect(printMb(1000 * MB, 2), '1000.00');
     });
+
+    test('msAsText', () {
+      expect(msAsText(3.111), equals('3.1 ms'));
+      expect(msAsText(3.111, includeUnit: false), equals('3.1'));
+      expect(msAsText(3.111, fractionDigits: 3), equals('3.111 ms'));
+      expect(msAsText(3), equals('3.0 ms'));
+    });
+
+    test('microsAsMsText', () {
+      expect(microsAsMsText(3111), equals('3.1 ms'));
+      expect(microsAsMsText(3111, includeUnit: false), equals('3.1'));
+      expect(microsAsMsText(3111, fractionDigits: 3), equals('3.111 ms'));
+      expect(microsAsMsText(3000), equals('3.0 ms'));
+    });
+
+    test('nullSafeMin', () {
+      expect(nullSafeMin(1, 2), equals(1));
+      expect(nullSafeMin(1, null), equals(1));
+      expect(nullSafeMin(null, 2), equals(2));
+      expect(nullSafeMin(null, null), equals(null));
+    });
+
+    test('nullSafeMin', () {
+      expect(nullSafeMax(1, 2), equals(2));
+      expect(nullSafeMax(1, null), equals(1));
+      expect(nullSafeMax(null, 2), equals(2));
+      expect(nullSafeMax(null, null), equals(null));
+    });
+
+    test('log2', () {
+      expect(log2(1), equals(0));
+      expect(log2(1.5), equals(0));
+      expect(log2(2), equals(1));
+      expect(log2(3), equals(1));
+      expect(log2(4), equals(2));
+    });
```

Comment:
thanks for adding tests for all this. Having good unit test and integration tests will only become more important as we get more users.